### PR TITLE
Refactor for v2

### PR DESCRIPTION
**Pull request by plasne**
This PR incorporates a number of significant refactors for v2, including:
- Removing WithMocks()
- Removing the ProvisionedResource RateLimiter
- Removing the explicit Stop()

### CODE REVIEW

**Comment by KeithJRome** [comment]
Was `docs/images/~$diagrams.pptx` included on accident?

File: README.md [hunk 1]
```diff
@@ -81,34 +81,30 @@ Batcher use cases are not limited to datastores. Consider the scenario where you
 
 - __Operation__: Operations are enqueued into the Batcher. An Operation has an associated "Watcher", a "cost", a designation of whether or not it can be batched, a counter for the number of times it has been "attempted", and a "payload" (which can be anything you want).
 
-There are also 2 rate limiters provided out-of-the-box...
+- __SharedResource__: This is a rate limiter that allows you to reserve a fixed amount of capacity and then share a fixed amount of capacity across multiple processes. An Azure Storage Account is used as the capacity lease management system when SharedCapacity is set.
```
Comment (KeithJRome) [inline README.md]: The second statement here is referring to the lease manager isn't it (as opposed to SharedResource itself)?

File: README.md [hunk 2]
```diff
@@ -81,34 +81,30 @@ Batcher use cases are not limited to datastores. Consider the scenario where you
...
 
-- __ProvisionedResource__: This is a simple rate limiter that has a fixed capacity per second.
-
-- __AzureSharedResource__: This rate limiter allows you to reserve a fixed amount of capacity and then share a fixed amount of capacity across multiple processes. An Azure Storage Account is used as the capacity lease management system.
+- __AzureBlobLeaseManager__: This is a Azure Blob Storage partition and lease management component used by SharedResource so that capacity can be shared across multiple instances.
 
 ![topology](./docs/images/topology.png)
```
Comment (KeithJRome) [inline README.md:89]: I don't think this diagram is still accurate?

**Comment by plasne** [comment]
> Was `docs/images/~$diagrams.pptx` included on accident?

Yes, I have removed it now. Thanks.

File: README.md [hunk 3]
```diff
@@ -120,9 +116,9 @@ Some other terms will be used throughout...
 
 - __Rate Limiting__: You may optionally attach a rate limiter to Batcher that can restrict the Operations so they don't exceed a certain cost per second.
 
-- __Shared Capacity__: Batcher supports using a rate limiter. One of the included rate limiters is AzureSharedResource which allows for sharing capacity across multiple Instances. Sharing capacity in this way can reduce cost.
+- __Shared Capacity__: Batcher supports using a rate limiter. One of the included rate limiters is SharedResource which allows for sharing capacity across multiple Instances. Sharing capacity in this way can reduce cost.
```
Comment (KeithJRome) [inline README.md]: There is only one rate limiter now, right? So "one of the included" would no longer be accurate?

File: README.md [hunk 4]
```diff
@@ -148,8 +144,8 @@ This code sample shows the general usage...
 
 1. If you are going to use rate limiting...
```
Comment (KeithJRome) [inline README.md:146]: It's zero or one rate limiter per batcher, right? This makes it sound like a batcher could/would use multiple limiters.

File: README.md [hunk 5]
```diff
@@ -246,11 +244,11 @@ There are several ways to facilitate unit testing when using Batcher. Detailed d
 
 ## Rate Limiting
 
-The AzureSharedResource rate limiter works like this...
+The SharedResource rate limiter works like this...
 
 ![lease](./docs/images/lease.png)
```
Comment (KeithJRome) [inline README.md:250]: This img is out of date

File: README.md [hunk 6]
```diff
@@ -274,15 +272,15 @@ To give a cost comparison with retail pricing in the East US region with 1 TB of
 
 ### Cost Increase
 
-Using default settings, each instance of AzureSharedResource will make a single storage transaction roughly every 250 milliseconds _when it needs additional capacity_. Therefore, we can determine the maximum cost for 4 instances on an Azure Storage Account GPv2 (or Blob) as...
+Using default settings, each instance of SharedResource will make a single storage transaction roughly every 250 milliseconds _when it needs additional capacity_. Therefore, we can determine the maximum cost for 4 instances on an Azure Storage Account GPv2 (or Blob) as...
```
Comment (KeithJRome) [inline README.md]: This should be referring to the lease manager for Azure, not the generic SharedResource that might only be configured for Reserved capacity

**Review by KeithJRome — CHANGES_REQUESTED** [review]
This is pretty much exactly how we talked about, but there's some corrections that need to be made to the docs.

File: README.md (hunk 5 above)
Comment (plasne) [reply README.md:250]: We did update the images, maybe they are cached. Are you seeing the one that says LeaseManager at the right?

Comment (KeithJRome) [reply README.md:250]: Very strange. I refreshed before and it showed the old versions, but now it shows the updated images. Must have been caching, but maybe server-side? Anyhow, I'll mark the image comments as resolved.

File: README.md (hunk 1 above)
Comment (norelina) [reply README.md]: Thanks @KeithJRome . Adjusted verbiage.

File: README.md (hunk 3 above)
Comment (norelina) [reply README.md]: Yes. Thanks @KeithJRome . Adjusted.

File: README.md (hunk 6 above)
Comment (norelina) [reply README.md]: Adjusted verbiage. Thanks @KeithJRome

File: README.md (hunk 4 above)
Comment (norelina) [reply README.md:146]: Adjusted wording to be clear. Thanks @KeithJRome

**Comment by HendoCode** [comment]
LGTM, now. Great work.

**Review by HendoCode — APPROVED** [review]
I'm looking forward to seeing more adoption of batches.

**Review by awkwardindustries — APPROVED** [review]
Feels like these changes really help strengthen alignment with desired usage. Great work to everyone thinking through ways to simplify this. Great to see the matching documentation, clear test cases... good stuff.

File: v2/azure-blob-lease-manager.go [hunk 7]
```diff
@@ -11,41 +11,37 @@ import (
...
-	mgr.parent = parent
 	return mgr
 }
 
-func (m *azureBlobLeaseManager) withMocks(container azureContainer, blob azureBlob) *azureBlobLeaseManager {
-	m.container = container
-	m.blob = blob
-	m.mocksInUse = true
-	return m
+// FOR INTERNAL USE ONLY. Events raised by AzureBlobLeaseManager must be raised to an Eventer. Specifically the SharedResource it is associated with
```
Comment (KeithJRome) [inline v2/azure-blob-lease-manager.go:37]: I think we agreed to remove these "FOR INTERNAL USE ONLY" tags

**Review by KeithJRome — COMMENTED** [review]
This looks good to me now! Thanks for incorporating yesterday's feedback.

**Pull request merged**